target_chains/ethereum/entropy_sdk/solidity/IEntropyConsumer.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: Apache 2
+pragma solidity ^0.8.0;
+
+abstract contract IEntropyConsumer {
+    // This method is called by Entropy to provide the random number to the consumer.
+    // It asserts that the msg.sender is the Entropy contract. It is not meant to be
+    // override by the consumer.
+    function _entropyCallback(uint64 sequence, bytes32 randomNumber) external {
+        address entropy = getEntropy();
+        require(entropy != address(0), "Entropy address not set");
+        require(msg.sender == entropy, "Only Entropy can call this function");
+
+        entropyCallback(sequence, randomNumber);
+    }
+
+    // getEntropy returns Entropy contract address. It is being used to check that the callback
```

Comment:
nit: suggest using `This method` instead of `It` in these doc comments -- being explicit adds clarity to the documentation. "it" can refer to so many different things that it can be confusing.